integration-tests/src/test/kotlin/org/stellar/anchor/platform/test/Sep24End2EndTests.kt

```diff
@@ -15,46 +15,80 @@ import org.stellar.walletsdk.ApplicationConfiguration
 import org.stellar.walletsdk.anchor.TransactionStatus
+import org.stellar.walletsdk.anchor.TransactionStatus.*
+import org.stellar.walletsdk.anchor.WithdrawalTransaction
 import org.stellar.walletsdk.asset.IssuedAssetId
 import org.stellar.walletsdk.auth.AuthToken
 import org.stellar.walletsdk.horizon.SigningKeyPair
+import org.stellar.walletsdk.horizon.sign
+import org.stellar.walletsdk.horizon.transaction.toTransferTransaction
 
 class Sep24End2EndTest(
-  private val config: TestConfig,
-  private val toml: Sep1Helper.TomlContent,
-  private val jwt: String
+    private val config: TestConfig,
+    private val toml: Sep1Helper.TomlContent,
+    private val jwt: String
 ) {
   private val walletSecretKey = System.getenv("WALLET_SECRET_KEY") ?: CLIENT_WALLET_SECRET
   private val keypair = SigningKeyPair.fromSecret(walletSecretKey)
   private val wallet =
-    Wallet(StellarConfiguration.Testnet, ApplicationConfiguration(useHttp = true))
+      Wallet(StellarConfiguration.Testnet, ApplicationConfiguration(useHttp = true))
   private val USDC =
-    IssuedAssetId("USDC", "GDQOE23CFSUMSVQK4Y5JHPPYK73VYCNHZHA7ENKCV37P6SUEO6XQBKPP")
+      IssuedAssetId("USDC", "GDQOE23CFSUMSVQK4Y5JHPPYK73VYCNHZHA7ENKCV37P6SUEO6XQBKPP")
   private val asset = USDC
   private val client =
-    HttpClient() {
-      install(HttpTimeout) {
-        requestTimeoutMillis = 300000
-        connectTimeoutMillis = 300000
-        socketTimeoutMillis = 300000
+      HttpClient() {
+        install(HttpTimeout) {
+          requestTimeoutMillis = 300000
+          connectTimeoutMillis = 300000
+          socketTimeoutMillis = 300000
+        }
       }
-    }
   private val anchor = wallet.anchor(config.env["anchor.domain"]!!.substring("http://".length))
-
   private val maxTries = 40
 
   private fun `typical deposit end-to-end flow`() = runBlocking {
     val token = anchor.auth().authenticate(keypair)
     // Start interactive deposit
     val deposit =
-      anchor.interactive().deposit(keypair.address, asset, token, mapOf("amount" to "10"))
+        anchor.interactive().deposit(keypair.address, asset, token, mapOf("amount" to "10"))
+    // Get transaction status and make sure it is INCOMPLETE
     val transaction = anchor.getTransaction(deposit.id, token)
-    assertEquals(TransactionStatus.INCOMPLETE, transaction.status)
-
+    assertEquals(INCOMPLETE, transaction.status)
+    // Make sure the interactive url is valid. This will also start the reference server's
+    // withdrawal process.
     val resp = client.get(deposit.url)
     print("accessing ${deposit.url}...")
     assertEquals(200, resp.status.value)
-    waitStatus(deposit.id, TransactionStatus.COMPLETED, token)
+    // Wait for the status to change to COMPLETED
+    waitStatus(deposit.id, COMPLETED, token)
+  }
+
+  private fun `typical withdraw end-to-end flow`() = runBlocking {
+    val token = anchor.auth().authenticate(keypair)
+    // TODO: Add the test where the amount is not specified
```

Comment:
You should add test endpoint or test function to be called from here (to provide amount) on the Ref server to support that

Reply:
The plan is to change them to parameterized tests and pass the amount here. The current design is to support running the tests from CLI. But it looks like we should run the tests as Junit tests from the CLI. 

https://stellarorg.atlassian.net/browse/ANCHOR-230